client/JSDebugger/jsDebugManager.ts

```diff
@@ -220,57 +225,73 @@ export class JsDebugManager {
+                    reject(mlxprsError);
                 });
         });
     }
 
-    public static async resolveDatabasetoId(username: string, password: string, database: string, hostname: string,
-        ssl?: boolean, ca?: Buffer, rejectUnauthorized = true, managePort = ClientContext.DEFAULT_MANAGE_PORT
-    ): Promise<string> {
+    public static async resolveDatabasetoId(database: string): Promise<string> {
+        const mlManageClientParameters = buildNewManageClientParametersFromConfig();
+        const manageClient = new ClientContext(mlManageClientParameters);
 
-        const url = ClientContext.buildUrl(hostname, '/v1/eval', ssl, managePort);
-        const script = `xdmp.database("${database}")`;
-        const options: Record<string, unknown> = {
-            headers: {
-                'Content-type': 'application/x-www-form-urlencoded',
-                'Accept': 'multipart/mixed',
-                'X-Error-Accept': ' application/json'
-            },
-            auth: {
-                user: username,
-                pass: password,
-                'sendImmediately': false
-            },
-            body: `javascript=${querystring.escape(script)}`
-        };
-        if (ca) options['agentOptions'] = { ca: ca };
-        options['rejectUnauthorized'] = rejectUnauthorized;
-        return request.post(url, options);
+        return new Promise((resolve, reject) => {
+            manageClient.databaseClient.internal.sendRequest(
+                '/v1/eval',
+                (requestOptions: ml.RequestOptions) => {
+                    requestOptions.method = 'POST';
+                    requestOptions.headers = {
+                        'Content-type': 'application/x-www-form-urlencoded',
+                        'Accept': 'multipart/mixed',
+                        'X-Error-Accept': ' application/json'
+                    };
+                },
+                (operation: ml.RequestOperation) => {
+                    operation.requestBody = `javascript=xdmp.database("${database}")`;
+                })
+                .result((result: object) => {
+                    const serverArray = JSON.parse(JSON.stringify(result));
+                    if (serverArray.length > 0) {
+                        resolve(serverArray[0].content);
+                    } else {
+                        resolve(null);
+                    }
+                })
+                .catch(() => {
+                    resolve(null);
+                });
+        });
     }
 
-    public static async getRequestInfo(username: string, password: string, requestId: string, debugServerName: string, hostname: string,
-        ssl?: boolean, ca?: Buffer, rejectUnauthorized = true, managePort = ClientContext.DEFAULT_MANAGE_PORT
-    ): Promise<string> {
+    public static async getRequestInfo(requestId: string, debugServerName: string): Promise<string> {
+        const mlManageClientParameters = buildNewManageClientParametersFromConfig();
```

Comment:
This block is in a few places - how about:

`const manageClient = newManageClient()`? 

Also, I think you can get away with `newManageConfig()` instead of `buildNewManageClientParametersFromConfig()`, as "manageConfig" is widely used in ml-app-deployer to represent the same concept.